### PR TITLE
Bug 1386054: enqueue in the trigger controller only when really needed

### PR DESCRIPTION
**Pull request by kargakis**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1386054

@mfojtik @smarterclayton @deads2k ptal

### CODE REVIEW

**Comment by kargakis** [comment]
[test]

File: pkg/deploy/controller/generictrigger/controller.go [hunk 1]
```diff
@@ -46,7 +46,9 @@ func (c *DeploymentTriggerController) Handle(config *deployapi.DeploymentConfig)
 }
 
 func (c *DeploymentTriggerController) handleErr(err error, key interface{}) {
-	if err == nil {
+	// TODO: "empty data" comes from the protobuf serializer when instantiate
+	// returns a 204. This should be a typed error we ignore in this controller.
```
Comment (mfojtik) [inline pkg/deploy/controller/generictrigger/controller.go]: can you create issue for this?

**Comment by mfojtik** [comment]
LGTM [merge]

File: pkg/deploy/controller/generictrigger/controller.go (hunk 1 above)
Comment (kargakis) [inline pkg/deploy/controller/generictrigger/controller.go]: https://github.com/kubernetes/kubernetes/issues/35126

**Pull request merged**